platform/iphone/game_center.mm

```diff
@@ -30,23 +30,10 @@
 
 #ifdef GAME_CENTER_ENABLED
 
-#include "game_center.h"
-
-#ifdef __IPHONE_9_0
-
-#import <GameKit/GameKit.h>
-extern "C" {
-
-#else
-
-extern "C" {
```

Comment:
I would really appreciate an explanation of the significance of `extern "C"` here. It doesn't seem to make a difference regardless, and was protected with `__IPHONE_9_0`. Since that's now our minimum deployment target, I thought it made sense to remove most of these guards.